lib/membrane_rtc_engine/endpoints/hls_endpoint.ex

```diff
@@ -279,7 +303,32 @@ if Enum.all?(
       link_builder = link_bin_input(pad)
       track = Map.get(state.tracks, track_id)
       directory = get_hls_stream_directory(state, track)
-      spec = hls_links_and_children(offset, link_builder, track, state, ctx)
+
+      {placements, video_layout} =
+        if is_nil(state.mixer_config) or track.type == :audio,
+          do: {[], state.video_layout},
+          else: state.video_layout_module.track_added(state.video_layout, track)
+
+      state = %{state | video_layout: video_layout}
+
+      initial_placements =
+        placements
+        |> Enum.filter(fn {Pad.ref(_type, id), _placement} ->
+          id == track.id or id == {:blank, track.id}
+        end)
+        |> Enum.map(fn
+          {Pad.ref(:input, {:blank, _id}), placement} -> {:blank, placement}
+          {Pad.ref(:input, _id), placement} -> {:input, placement}
+        end)
+        |> Map.new()
+
+      placements =
+        placements
+        |> Enum.filter(fn {Pad.ref(_type, id), _placement} ->
+          id != track.id and id != {:blank, track.id}
```

Comment:
This condition is a negation condition from line 317, could we move it to some support function? Here, we could use the same condition as in the line above and function `Enum.reject`.